src/str_replace.c

```diff
@@ -20,29 +22,43 @@ char *str_replace(const char *orig, const char *rep, const char *with) {
         with = "";
     len_with = strlen(with);
 
+    orig_copy = malloc(sizeof(char) * (strlen(orig) + 1));
+
+    if (!orig_copy) {
+        return NULL;
+    }
+
+    strcpy(orig_copy, orig);
```

Comment:
Since you’re changing this to avoid a race condition, `strncpy` with the limit of allocated memory would be more suitable here.